### PR TITLE
repo: update typescript references

### PR DESCRIPTION
**Pull request by marechal-p**
Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

#### What it does

1. Update typescript references.
2. Add a CI script to check if typescript references are in sync.

#### How to test

- Run `yarn test:references`: it should work.
- Change a reference in any package to something bogus (non-existant path).
- Run `yarn test:references`: it should fail.

### CODE REVIEW

**Comment by marechal-p** [comment]
Not sure if we should automate the script to compile references?

**Comment by vince-fugnitto** [comment]
> Not sure if we should automate the script to compile references?

I think so, how will we know if a reference is missing? (how did we know in this case)

**Comment by marechal-p** [comment]
@vince-fugnitto indeed, added a script to check everything is in sync while running the tests. People will need to manually trigger the update script to make sure everything is in order if the check fails.

**Comment by akosyakov** [comment]
Why don't run references generation during prepare always, similarly how we prepare travis file when new project is added?

**Comment by marechal-p** [comment]
@akosyakov the reason is that I believe human supervision is required should the script run. The script guaranties that our workspace layout it followed by TypeScript, but we can also add specific references that are out of this system and I'm "afraid" for those. Also I don't like scripts that crawl the disk to run when not needed, especially when they can modify your workspace.

**Comment by marechal-p** [comment]
Updated, fingers crossed that the script will behave. Worst case ping me if something goes wrong.

**Comment by akosyakov** [comment]
> the reason is that I believe human supervision is required should the script run

@marechal-p we can supervise output of a script and fix the script to behave

**Comment by marechal-p** [comment]
@akosyakov that's done. The script now sets all ts references between our local repos, and only that. People shouldn't edit the references manually as it might get overridden by the script. Should someone think we need a different set of references, the script would have to be updated. Is this ok?

**Comment by akosyakov** [comment]
> Should someone think we need a different set of references, the script would have to be updated. Is this ok?

yes, i think it is fine, we should discuss if someone wants something like that anyway, thank you

File: scripts/compile-references.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// @ts-check
```
Comment (akosyakov) [inline scripts/compile-references.js]: we should have copyright here?

Comment (akosyakov) [reply scripts/compile-references.js]: ah, see it is below, not sure what it is correct, but anyway

**Comment by akosyakov** [comment]
looks good to me, @marcdumais-work @vince-fugnitto could you have a look as well?

File: scripts/compile-references.js (hunk 1 above)
Comment (marechal-p) [reply scripts/compile-references.js]: `// @ts-check` has to be at the top for it to work apparently

**Comment by marechal-p** [comment]
@vince-fugnitto I just improved logging a bit, thanks to your output. I will now rebase, run the scripts again to see if anything needs updating, and finally merge.

**Pull request merged**